polars/polars-io/src/csv/splitfields.rs

```diff
@@ -0,0 +1,327 @@
+#[cfg(not(feature = "simd"))]
+mod inner {
+    /// An adapted version of std::iter::Split.
+    /// This exists solely because we cannot split the lines naively as
+    pub(crate) struct SplitFields<'a> {
+        v: &'a [u8],
+        delimiter: u8,
+        finished: bool,
+        quote_char: u8,
+        quoting: bool,
+        eol_char: u8,
+    }
+
+    impl<'a> SplitFields<'a> {
+        pub(crate) fn new(
+            slice: &'a [u8],
+            delimiter: u8,
+            quote_char: Option<u8>,
+            eol_char: u8,
+        ) -> Self {
+            Self {
+                v: slice,
+                delimiter,
+                finished: false,
+                quote_char: quote_char.unwrap_or(b'"'),
+                quoting: quote_char.is_some(),
+                eol_char,
+            }
+        }
+
+        unsafe fn finish_eol(
+            &mut self,
+            need_escaping: bool,
+            idx: usize,
+        ) -> Option<(&'a [u8], bool)> {
+            self.finished = true;
+            debug_assert!(idx <= self.v.len());
+            Some((self.v.get_unchecked(..idx), need_escaping))
+        }
+
+        fn finish(&mut self, need_escaping: bool) -> Option<(&'a [u8], bool)> {
+            self.finished = true;
+            Some((self.v, need_escaping))
+        }
+
+        fn eof_oel(&self, current_ch: u8) -> bool {
```

Comment:
Typo in function name?

Reply:
Why? End of field/ end of line

Reply:
> Why? End of field/ end of line

Yes, the letters for "end of line" are mixed up. Currently it means "End of field/Of end line". 'oel' -> 'eol'